File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2166.cpp
#include <iostream>

using namespace std;

double fracao(double n){
    if(n==0){
        return 0; 
    } else if(n==1){
        return 1/2.0;
    }
    return 1.0/(2+fracao(n-1));
}

int main() {
    int n;
    cin >> n;
    
    cout.precision(10);
    cout.setf(ios::fixed);

    cout << 1.0+fracao(n) << endl;
	
    return 0;
}

File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2166.py
def fracao(n):
    if n == 0:
        return 0
    elif n == 1:
        return 1 / 2.0
    return 1 / (2 + fracao(n-1))
    
def main():
    print("{:.10f}".format(1.0 + fracao(int(input()))))
    
main()